host_applications/linux/apps/raspicam/RaspiCommonSettings.c

```diff
@@ -0,0 +1,218 @@
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+ * \file RaspiCommonSettings.c
+ *
+ * \Author: James Hughes
+ *
+ * Description
+ *
+ * Handles general settings applicable to all the camera applications
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+#include <memory.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sysexits.h>
+
+#include "interface/vcos/vcos.h"
+#include "interface/mmal/mmal.h"
+#include "interface/mmal/mmal_logging.h"
+#include "interface/mmal/mmal_buffer.h"
+#include "interface/mmal/util/mmal_util.h"
+#include "interface/mmal/util/mmal_util_params.h"
+#include "interface/mmal/util/mmal_default_components.h"
+#include "interface/mmal/util/mmal_connection.h"
+#include "interface/mmal/mmal_parameters_camera.h"
+
+#include "RaspiCommonSettings.h"
+#include "RaspiCLI.h"
+#include "RaspiHelpers.h"
+
+enum
+{
+   CommandHelp,
+   CommandWidth,
+   CommandHeight,
+   CommandOutput,
+   CommandVerbose,
+   CommandCamSelect,
+   CommandSensorMode,
+};
+
+
+static COMMAND_LIST cmdline_commands[] =
+{
+   { CommandHelp,    "-help",       "?",  "This help information", 0 },
+   { CommandWidth,   "-width",      "w",  "Set image width <size>", 1 },
+   { CommandHeight,  "-height",     "h",  "Set image height <size>", 1 },
+   { CommandOutput,  "-output",     "o",  "Output filename <filename> (to write to stdout, use '-o -'). If not specified, no file is saved", 1 },
+   { CommandVerbose, "-verbose",    "v",  "Output verbose information during run", 0 },
+   { CommandCamSelect, "-camselect","cs", "Select camera <number>. Default 0", 1 },
+   { CommandSensorMode,"-mode",     "md", "Force sensor mode. 0=auto. See docs for other modes available", 1},
+};
+
+static int cmdline_commands_size = sizeof(cmdline_commands) / sizeof(cmdline_commands[0]);
+
+
+void raspicommonsettings_set_defaults(RASPICOMMONSETTINGS_PARAMETERS *state)
+{
+   strncpy(state->camera_name, "(Unknown)", MMAL_PARAMETER_CAMERA_INFO_MAX_STR_LEN);
+   // We dont set width and height since these will be specific to the app being built.
+   state->width = 0;
+   state->height = 0;
+   state->filename = NULL;
+   state->verbose = 0;
+   state->cameraNum = 0;
+   state->sensor_mode = 0;
+};
+
+/**
+ * Dump parameters as human readable to stdout
```

Comment:
Erm, stdout, or stderr.
I thought we had previously changed it so help text went to stdout so that people could ```less``` and ```grep``` it. Verbose text went to stderr so that it didn't interfere with ```-o -``` and piping the output.

Reply:
Correct, C&P error from another file, that presumably has the same error which I will try and find.